packages/components/src/date-time/time/index.tsx

```diff
@@ -165,6 +170,50 @@ export function TimePicker( {
 		</MonthSelectWrapper>
 	);
 
+	const yearField = (
+		<YearInput
+			key="year"
+			className="components-datetime__time-field components-datetime__time-field-year" // Unused, for backwards compatibility.
+			label={ __( 'Year' ) }
+			hideLabelFromVision
+			__next40pxDefaultSize
+			value={ year }
+			step={ 1 }
+			min={ 1 }
+			max={ 9999 }
+			required
+			spinControls="none"
+			isPressEnterToChange
+			isDragEnabled={ false }
+			isShiftStepEnabled={ false }
+			onChange={ buildNumberControlChangeCallback( 'year' ) }
+			__unstableStateReducer={ buildPadInputStateReducer( 4 ) }
+		/>
+	);
+
+	let dateOrder;
+
+	if ( ! _dateOrder ) {
+		dateOrder = is12Hour ? 'mdy' : 'dmy';
+	} else {
+		dateOrder = VALID_DATE_ORDERS.includes( _dateOrder )
+			? _dateOrder
+			: DEFAULT_DATE_ORDER;
+	}
```

Comment:
Here's a suggestion to simplify this logic a bit:

```diff
diff --git a/packages/components/src/date-time/time/index.tsx b/packages/components/src/date-time/time/index.tsx
index b09e35fee7..e2877a1ded 100644
--- a/packages/components/src/date-time/time/index.tsx
+++ b/packages/components/src/date-time/time/index.tsx
@@ -34,7 +34,6 @@ import {
 import { TIMEZONELESS_FORMAT } from '../constants';
 import { TimeInput } from '../time-input';
 
-const DEFAULT_DATE_ORDER = 'dmy';
 const VALID_DATE_ORDERS = [ 'dmy', 'mdy', 'ymd' ];
 
 /**
@@ -61,7 +60,7 @@ export function TimePicker( {
 	is12Hour,
 	currentTime,
 	onChange,
-	dateOrder: _dateOrder,
+	dateOrder: dateOrderProp,
 }: TimePickerProps ) {
 	const [ date, setDate ] = useState( () =>
 		// Truncate the date at the minutes, see: #15495.
@@ -191,15 +190,11 @@ export function TimePicker( {
 		/>
 	);
 
-	let dateOrder;
-
-	if ( ! _dateOrder ) {
-		dateOrder = is12Hour ? 'mdy' : 'dmy';
-	} else {
-		dateOrder = VALID_DATE_ORDERS.includes( _dateOrder )
-			? _dateOrder
-			: DEFAULT_DATE_ORDER;
-	}
+	const defaultDateOrder = is12Hour ? 'mdy' : 'dmy';
+	const dateOrder =
+		dateOrderProp && VALID_DATE_ORDERS.includes( dateOrderProp )
+			? dateOrderProp
+			: defaultDateOrder;
 
 	const fields = dateOrder.split( '' ).map( ( field ) => {
 		switch ( field ) {
```

The main points being:

- Rename `_dateOrder` to `dateOrderProp` — The underscore is fine, but this is just a convention we've been using in this package when renaming props like this.
- Remove the `DEFAULT_DATE_ORDER` variable.
- Simply fall back to the 12-hour conditional logic whenever `dateOrderProp` is undefined or invalid.